skimage/filters/thresholding.py

```diff
@@ -766,27 +766,36 @@ def _mean_std(image, w):
            Retrieval XV, (San Jose, USA), Jan. 2008.
            DOI:10.1117/12.767755
     """
-    if w == 1 or w % 2 == 0:
+    if isinstance(w, int):
+        w1 = w
+        w2 = w
+        kern = np.zeros((w2 + 1,) * image.ndim)
```

Comment:
I think you should normalise all the inputs to be the same. So, if you're given a window size of 5, with a 3D image, you want to make w equal to (6, 6, 6). Or if it's a 2D image and you're given a kernel of (3, 5), you make w equal to (4, 6). And so on. Then you work as if every input was a correct tuple.